internal/service/connect/phone_number.go

```diff
@@ -0,0 +1,292 @@
+				ValidateFunc: validation.StringInSlice(connect.PhoneNumberCountryCode_Values(), false),
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 500),
+			},
+			"phone_number": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"prefix": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validPhoneNumberPrefix,
+			},
+			"status": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"message": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"target_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(connect.PhoneNumberType_Values(), false),
+			},
+			"tags":     tftags.TagsSchema(),
+			"tags_all": tftags.TagsSchemaComputed(),
+		},
+	}
+}
+
+func resourcePhoneNumberCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).ConnectConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	targetArn := d.Get("target_arn").(string)
+	phoneNumberType := d.Get("type").(string)
+
+	input := &connect.SearchAvailablePhoneNumbersInput{
+		MaxResults:             aws.Int64(1),
+		PhoneNumberCountryCode: aws.String(d.Get("country_code").(string)),
+		PhoneNumberType:        aws.String(phoneNumberType),
+		TargetArn:              aws.String(targetArn),
+	}
+
+	if v, ok := d.GetOk("prefix"); ok {
+		input.PhoneNumberPrefix = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Searching for Connect Available Phone Numbers %s", input)
+	output, err := conn.SearchAvailablePhoneNumbersWithContext(ctx, input)
+
+	if err != nil {
+		return diag.Errorf("searching Connect Phone Number for Connect Instance (%s,%s): %s", targetArn, phoneNumberType, err)
+	}
+
+	if output == nil || output.AvailableNumbersList == nil || len(output.AvailableNumbersList) == 0 {
+		return diag.Errorf("searching Connect Phone Number for Connect Instance (%s,%s): empty output", targetArn, phoneNumberType)
+	}
+
+	phoneNumber := output.AvailableNumbersList[0].PhoneNumber
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("generating uuid for ClientToken for Connect Instance (%s,%s): %s", targetArn, aws.StringValue(phoneNumber), err)
+	}
+
+	input2 := &connect.ClaimPhoneNumberInput{
+		ClientToken: aws.String(uuid), // can't use aws.String(resource.UniqueId()), because not a valid uuid
```

Comment:
I don't believe there's a required UUID format for ClientToken: https://docs.aws.amazon.com/connect/latest/APIReference/API_ClaimPhoneNumber.html#API_ClaimPhoneNumber_RequestSyntax 

As long as it's less than 500 characters.